api/specs.proto

```diff
@@ -341,6 +341,11 @@ message NetworkSpec {
 	// swarm internally created only and it was identified by the name
 	// "ingress" and the label "com.docker.swarm.internal": "true".
 	bool ingress = 7;
+
+	// ConfigFrom indicates that the network specific configuration
+	// for this network will be provided via another network, locally
+	// on the node where this network is being plumbed.
+	string config_from = 8;
```

Comment:
@aluzzardi this is the alternate proposal / implementation for https://github.com/docker/swarmkit/issues/1477. As mentioned by @mgoelzer (https://github.com/docker/swarmkit/issues/1477#issuecomment-249734558), this is a popular request for users of macvlan/ipvlan or other node-local drivers which has node-level configuration management, which cannot be in a sane way managed from the manager.

Also swapping the allocator and scheduling stage as described in https://github.com/docker/swarmkit/issues/1477 will not work in many cases (including the most common macvlan which might require a different parent interface per worker node).

Also, as discussed in the proposal raised by @ijc25 , this will help with a similar story of swarmkit not involving in the actual allocation of network resources and letting the drivers on the worker nodes to handle network resources.